students/Josh_HOff/lesson09/mailroom_lesson_09.py

```diff
@@ -4,70 +4,149 @@
 
 donors = {'Josh Hoff': [25, 75], 'Tatsiana Kisel': [35, 105.55]}
 
+#This class handles all input from the user
+class Input(object):
+
+#    donors = {'Josh Hoff': [25, 75], 'Tatsiana Kisel': [35, 105.55]}
 
-class Donor(object):
     def __init__(self=None):
         pass
-        
+    
     def thank_you(self=None):
         while True:
             donor_name = input('\nWhat is the name of the donor?: ')
             if donor_name == 'list':
-                Donor().show_list()
+                DonorCollection().show_list()
                 continue
+            elif donor_name == 'quit':
+                break
             while True:
                 try:
                     donation = float(input('\nWhat is the donation amount?: '))
                 except ValueError:
                     print('\nPlease give a number instead.')
                     continue
                 break
-            Donor().add_donor(donor_name, donation)
+            DonorCollection().add_donor(donor_name, donation)
             break
+            
+    def quitting(self):
+        sys.exit()
+        
+    def continuing(self):
+        print('Try Again.\n')
+
+#this class handles all donor management
+class Donor(object):
+    def __init__(self=None, name = ''):
+        global donors
+        self._donors = donors
+        self._gifts = len(self._donors.get(name))
+        self._total = sum(self._donors.get(name))
+        self._average = round((self._total / self._gifts), 2)
+        self._recent_gift = self._donors.get(name)[-1]
+        self._first_gift = self._donors.get(name)[0]
+                
+    @property
+    def gifts(self):
+        return self._gifts
+        
+    @gifts.deleter
+    def gifts(self):
+        del self._gifts
+        
+    @property
+    def total_donations(self):
+        return self._total
+        
+    @total_donations.deleter
+    def total_donations(self):
+        del self._total
+        
+    @property
+    def average(self):
+        return self._average
+        
+    @average.deleter
+    def average(self):
+        del _average
+        
+    @property
+    def recent_gift(self):
+        return self._recent_gift
+        
+    @recent_gift.deleter
+    def recent_gift(self):
+        del _recent_gift
+        
+    @property
+    def first_gift(self):
+        return self._first_gift
+        
+    @first_gift.deleter
+    def first_gift(self):
+        del self._first_gift
+        
+        
+class DonorCollection(object):
+    def __init__(self=None):
+        pass
```

Comment:
this isn't really object oriented, you're not holding any state? 

staticmethods are just helper functions that relate to a class, not necessary part of an object.